test/adapter/polymorphic_test.rb

```diff
@@ -20,31 +30,117 @@ def test_attributes_serialization
               id: 1,
               title: 'headshot-1.jpg',
               imageable: {
-                id: 42,
-                name: 'Zoop Zoopler'
+                  type: 'employee',
+                  employee: {
+                    id: 42,
+                    name: 'Zoop Zoopler'
+                  }
               }
             }
 
-          assert_equal(expected, @attributes_serialization.as_json)
+          assert_equal(expected, serialization(@picture))
         end
 
-        def test_json_serializer
+        def test_attributes_serialization_without_polymorphic_association
+          expected =
+            {
+              id: 2,
+              title: 'headshot-2.jpg',
+              imageable: nil
+            }
+
+          assert_equal(expected, serialization(@simple_picture))
+        end
+
+        def test_attributes_serialization_with_polymorphic_has_many
```

Comment:
@beauby 'heterogeneous has-many" tests for :json and :attribute adapters added.  If my understanding was correct, all the plumbing of the Attributes class acted as expected, tests demonstrate the nesting (which I feel is in alignment with previous versions [https://github.com/rails-api/active_model_serializers/pull/1726#issuecomment-219323927])